lib/shelley/src/Cardano/Wallet/Shelley/Compatibility.hs

```diff
@@ -1514,10 +1556,36 @@ fromAlonzoTxBodyAndAux bod mad =
         _network
         = bod
     (assetsToMint, assetsToBurn) = fromLedgerMintValue mint
+    scriptMap = fromAlonzoScriptMap $ Alonzo.txscripts' wits
+    (mintScriptMap, burnScriptMap)
+        | (assetsToMint == TokenMap.empty && assetsToBurn /= TokenMap.empty)
+        = (Map.empty, scriptMap)
+        | (assetsToMint /= TokenMap.empty && assetsToBurn == TokenMap.empty)
+        = (scriptMap, Map.empty)
+        | otherwise = (Map.empty, Map.empty)
```

Comment:
@paweljakubas What if the `mint` field contains **_both_** assets to mint **_and_** assets to burn?

It seems that in this case, we'd fall through to the `otherwise` case.

Is that expected?

According to my understanding, it's possible to **_both_** mint **_and_** burn assets in a single transaction:
- Assets to **_mint_** are represented by **_positive_** values in the `mint` map.
- Assets to **_burn_** are represented by **_negative_** values in the `mint` map.

The `fromLedgerMintValue` function separates these out into two `TokenMap` values, because `TokenMap` values can only contain non-negative quantities.

Reply:
added to investigate that in next PR: